### PR TITLE
mautrix-telegram: 0.11.3 -> 0.12.0 and rework packaging

### PR DESCRIPTION
**Pull request by NickCao**
###### Description of changes
Removed some unused dependencies, added some optional ones. `moviepy` is left out by default as it greatly bloats the closure size by 600M.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by NickCao** [comment]
cc @yu-re-ka

**Comment by NickCao** [comment]
now depends on https://github.com/NixOS/nixpkgs/pull/185497

**Comment by NickCao** [comment]
Result of `nixpkgs-review pr 183261` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>mautrix-signal</li>
  </ul>
</details>
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>mautrix-facebook</li>
    <li>mautrix-telegram</li>
    <li>python310Packages.cryptg</li>
    <li>python310Packages.mautrix (python310Packages.mautrix-appservice)</li>
    <li>python39Packages.cryptg</li>
    <li>python39Packages.mautrix (python39Packages.mautrix-appservice)</li>
  </ul>
</details>

**Comment by NickCao** [comment]
Result of `nixpkgs-review pr 183261` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>mautrix-facebook</li>
    <li>mautrix-signal</li>
    <li>mautrix-telegram</li>
    <li>python310Packages.cryptg</li>
    <li>python310Packages.mautrix (python310Packages.mautrix-appservice)</li>
    <li>python39Packages.cryptg</li>
    <li>python39Packages.mautrix (python39Packages.mautrix-appservice)</li>
  </ul>
</details>

File: pkgs/servers/mautrix-telegram/default.nix [hunk 1]
```diff
@@ -1,95 +1,73 @@
...
     pycryptodome
     unpaddedbase64
-  ]) ++ dbDrivers;
+  ]);
 
-  # Tests are broken and throw the following for every test:
-  #   TypeError: 'Mock' object is not subscriptable
-  #
-  # The tests were touched the last time in 2019 and upstream CI doesn't even build
-  # those, so it's safe to assume that this part of the software is abandoned.
```
Comment (SuperSandro2000) [inline pkgs/servers/mautrix-telegram/default.nix:80]: Does this comment still apply? If so I would like to at least add back that the tests are not functional/broken.

File: pkgs/development/python-modules/cryptg/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    hash = "sha256-M2ySVqfgpgHktLh4t5Sh1UTBCzajlQiDku4O9azHJwk=";
+  };
+
+  nativeBuildInputs = with rustPlatform;[
+    setuptools-rust
+    cargoSetupHook
+    rust.rustc
+    rust.cargo
+  ];
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/cryptg/default.nix:31]:
```suggestion

  # has no tests
  doCheck = false;

```

File: pkgs/servers/mautrix-telegram/default.nix (hunk 1 above)
Comment (NickCao) [reply pkgs/servers/mautrix-telegram/default.nix:80]: The tests just do not exists now, they were completely removed.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 183261` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
7 packages built:
</summary>
<ul>
<li>mautrix-facebook</li>
<li>mautrix-signal</li>
<li>mautrix-telegram</li>
<li>python310Packages.cryptg</li>
<li>python310Packages.mautrix (python310Packages.mautrix-appservice)</li>
<li>python39Packages.cryptg</li>
<li>python39Packages.mautrix (python39Packages.mautrix-appservice)</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
mautrix-facebook:
<details>

got build log for '/nix/store/iga51gppabpl1vgza22ja88y36wf27y4-python3.10-mautrix-facebook-unstable-2022-05-06' from 'daemon'
When evaluating attribute ‘mautrix-facebook’:
warning: unused-argument
Unused argument: `fetchpatch`.
Near pkgs/servers/mautrix-facebook/default.nix:3:3:

```
  |
3 | , fetchpatch
  |   ^
```
</details>
mautrix-signal:
<details>

got build log for '/nix/store/p87prdp3fbv753174ndxcll3pgdxknwh-python3.10-mautrix-signal-0.3.0' from 'daemon'
When evaluating attribute ‘mautrix-signal’:
warning: python-include-tests
Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/servers/mautrix-signal/default.nix:56:

```
   |
56 |     description = "A Matrix-Signal puppeting bridge";
   | ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
mautrix-telegram:
<details>

got build log for '/nix/store/idpny4idq896svqngsrcb7y4vv2yh9m9-python3.10-mautrix-telegram-0.12.0' from 'daemon'
When evaluating attribute ‘mautrix-telegram’:
warning: unused-argument
Unused argument: `mautrix-telegram`.
Near pkgs/servers/mautrix-telegram/default.nix:3:3:

```
  |
3 | , mautrix-telegram
  |   ^
```
</details>
</details>

**Pull request merged**